synchro/__init__.py

```diff
@@ -668,7 +675,11 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        self.close()
+        if exc_type is None:
+            self.close()
+        else:
+            object.__setattr__(self, "_closed", True)
+        return False
```

Comment:
We should try to rely on the motor method here otherwise we're testing synchro itself and not actually testing motor. Perhaps:
```
return self.synchronize(self.delegate.__aexit__)()
```